src/ape_ethereum/ecosystem.py

```diff
@@ -345,7 +345,11 @@ def encode_calldata(self, abi: Union[ConstructorABI, MethodABI], *args) -> HexBy
         parser = StructParser(abi)
         arguments = parser.encode_input(args)
         input_types = [i.canonical_type for i in abi.inputs]
-        converted_args = self.conversion_manager.convert(arguments, tuple)
+
+        # TODO: Convert input_types to python_types. e.g. uint256 -> int, bytes4 -> HexBytes etc.
```

Comment:
@Ninjagod1251 I saved the hardest part for you to figure out!
We need to derive the Pythonic types from the ABI input types here so we can pass them into the conversion manager.

Like `bytes4` should be `HexBytes`
`address` -> `AddressType`
etc.